### ISSUE TITLE
[New metric] Add `CharacTER`

### ISSUE THREAD

**Issue by stancld** [PyTorchLightning/metrics#633]
## 🚀 Feature

Add `CharacTER`,  a text metric used for NMT evaluation.

**Sources:**
Paper - [CharacTER: Translation Edit Rate on Character Level](https://aclanthology.org/W16-2342.pdf)
[Repo](https://github.com/rwth-i6/CharacTER)

### Motivation / Abstract

Recently, the capability of character-level evaluation measures for machine translation output has been confirmed by several metrics. This work proposes translation edit rate on character level (CharacTER), which calculates the character level edit distance while performing the shift edit on word level. The novel metric shows high system-level correlation with human rankings, especially for morphologically rich languages. It outperforms the strong CHRF by up to 7% correlation on different metric tasks. In addition, we apply the hypothesis sentence length for normalizing the edit distance in CharacTER, which also provides significant improvements compared to using the reference sentence length. ([Wang et al., 2016](https://aclanthology.org/W16-2342.pdf))

**Comment by Rajathbharadwaj** [comment]
Hey @stancld @SkafteNicki, I want to contribute, how do I get started? Thanks.

**Comment by stancld** [comment]
Hi @Rajathbharadwaj, I hope you can have a look at #641 and try to prepare a PR for `CharacTER` accordingly. Once you have anything ready, you can open a draft PR and ping us :]

**Comment by SkafteNicki** [comment]
Hi @Rajathbharadwaj, agree with @stancld that PR #641 I good place to look what how a complete implementation (+ docs ) should be structured. If you do not have time to fully implement it all, feel free to also open a PR with a partial implementation and then we can guide you to the rest :]

**Comment by Rajathbharadwaj** [comment]
Okay, I will check out and hopefully open a draft PR. Thanks!

**Comment by Rajathbharadwaj** [comment]
@SkafteNicki @stancld So I did check out the PR #641 and I can see there are some helper functions or some extra functions compared to the source repo. It'd be great if I could get some help in those regards. I will go ahead and create the module now. Thanks!

**Comment by stancld** [comment]
Hello @Rajathbharadwaj, any update on this here? Let me know if you need any help :]

**Comment by stancld** [comment]
Hi @Rajathbharadwaj, any updates? :]

**Comment by Rajathbharadwaj** [comment]
Hey sorry about the delay. I will start contributing, been a bit busy.

Thanks.

**Comment by Borda** [comment]
> Hey sorry about the delay. I will start contributing, been a bit busy.

no pressure, take your time :rabbit:

**Comment by ankitaS11** [comment]
Hey @stancld @SkafteNicki, I want to pick this issue, if its fine with everyone.

**Comment by stancld** [comment]
@ankitaS11 Yes, it's fine to take this one, but please be sure #855 to be merged before diving into this one :]

**Comment by ankitaS11** [comment]
Sure. Will make that PR ready for review by this weekend.

**Comment by stancld** [comment]
Hi @ankitaS11, any updates here? :]
